lms/djangoapps/ora_staff_grader/views.py

```diff
@@ -82,14 +108,30 @@ def submit_grade(self, request, usage_id, grade_data):
         Returns: {'success': True/False, 'msg': err_msg}
         """
         response = call_xblock_json_handler(request, usage_id, 'staff_assess', grade_data)
-        return json.loads(response.content)
+
+        # Unhandled errors might not be JSON, catch before loading
+        if response.status_code != 200:
+            raise Exception()
+
+        response_data = json.loads(response.content)
+
+        # Handled faillure still returns HTTP 200 but with 'success': False and supplied error message "msg"
+        if not response_data.get('success', False):
+            raise ExceptionWithContext(context={"msg": response_data.get('msg', '')})
+
+        return response_data
 
     def check_submission_lock(self, request, usage_id, submission_uuid):
         """
         Look up lock info for the given submission by calling the ORA's 'check_submission_lock' XBlock.json_handler
         """
         data = {'submission_uuid': submission_uuid}
         response = call_xblock_json_handler(request, usage_id, 'check_submission_lock', data)
+
+        # Unclear that there would every be an error (except network/auth) but good to catch here
+        if response.status_code != 200:
+            raise Exception()
```

Comment:
I think this should be a custom error we're raising in all these places so we can avoid the flat `except Exception:` in the views. We want to know when this scenario happens, but other exceptions that may be raised should probably still bubble up so we can catch them in tools like NewRelic.